omnipaxos/src/ballot_leader_election.rs

```diff
@@ -75,12 +75,14 @@ pub(crate) struct BallotLeaderElection {
     /// The current round of the heartbeat cycle.
     hb_round: u32,
     /// Vector which holds all the received heartbeats
-    ballots: Vec<(Ballot, Connectivity)>,
+    ballots: Vec<(Ballot, Connectivity, bool)>,
     /// Holds the current ballot of this instance.
     current_ballot: Ballot,
     /// The number of replicas inside the cluster that this instance is
     /// connected to (based on heartbeats received) including itself.
     connectivity: Connectivity,
+    happy: bool,
+    gossiped_leader: Ballot,
```

Comment:
I think a better name is `highest_seen_ballot` since it can be set by ourselves (and therefore not necessarily gossiped) and can get updated before it is confirmed as a leader.